src/frontend/test_runner/tests/testdata/subquery.yaml

```diff
@@ -114,7 +114,7 @@
           LogicalJoin { type: Inner, on: true, output_indices: all }
             LogicalScan { table: bc, columns: [_row_id, b, c] }
             LogicalProject { exprs: [$1, $2] }
-              LogicalFilter { predicate: ($1 = CorrelatedInputRef { index: 1, depth: 2 }) }
+              LogicalFilter { predicate: ($1 = CorrelatedInputRef { index: 1, correlated_id: 0 }) }
```

Comment:
So we need to do more than #4085 to fix this ...

Reply:
Yes, this is the problem of binder, because of correlated_id = 0 means we fail to bind correlated_input_ref to its `Apply` by `depth`.